WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt

```diff
@@ -0,0 +1,143 @@
+            options.isEnableSessionTracking = true // Release Health tracking
+            options.setBeforeSend { event, _ ->
+
+                if (!this.shouldSendEvents(context)) return@setBeforeSend null
+
+                if (event.exceptions.size > 1) {
+                    event.exceptions.lastOrNull()?.let { lastException ->
+                        // Remove the "Invoking subscriber failed" exception so that the main error will show up
+                        // in Sentry. This error only means that an exception occurred during an EventBus event and
+                        // it's not particularly useful for debugging.
+                        if (lastException.module == EVENT_BUS_MODULE &&
+                                lastException.type == EVENT_BUS_EXCEPTION &&
+                                lastException.value == EVENT_BUS_INVOKING_SUBSCRIBER_FAILED_ERROR) {
+                            event.exceptions.remove(lastException)
+                        }
+                    }
+                }
+                event
+            }
+        }
+
+        Sentry.setTag("version", BuildConfig.VERSION_NAME)
+
+        this.accountStore.account.apply {
+            val user = User()
+            user.id = this.userId.toString()
+            user.username = this.userName
+            user.email = this.email
+
+            Sentry.setUser(user)
+        }
+    }
+
+    /**
+     * Logs a message as a Sentry breadcrumb
+     *
+     * This doesn't generate a Sentry event – it just records it as context for any future crashes. To
+     * generate a Sentry event, use [report].
+     *
+     * @param[message] The message to log
+     * @param[tag] An optional [AppLog] tag
+     */
+    @JvmOverloads
+    fun log(message: String, tag: AppLog.T? = null) = tag?.let {
+        Sentry.addBreadcrumb(message, tag.toString())
+    } ?: run {
+        Sentry.addBreadcrumb(message)
+    }
+
+    /**
+     * Logs as an exception as a Sentry Breadcrumb
+     *
+     * This doesn't generate a Sentry event – it just records it as context for any future crashes. To
+     * generate a Sentry event, use [reportException].
+     *
+     * @param[message] The message to log
+     * @param[tag] An optional [AppLog] tag
+     */
+    @JvmOverloads
+    fun logException(tr: Throwable, tag: AppLog.T? = null) = this.log(tr.toString(), tag)
+
+    /**
+     * Send a message to Sentry as a new event
+     *
+     * @param[message] The message to log
+     * @param[tag] An optional [AppLog] tag
+     */
+    @JvmOverloads
+    fun report(message: String, tag: AppLog.T? = null) {
+        if (tag != null) {
+            Sentry.setExtra("tag", tag.toString())
+            Sentry.setTag("tag", tag.toString())
+        }
+
+        val sentryId = Sentry.captureMessage(message)
+        AppLog.d(T.UTILS, "Captured Sentry Event: $sentryId")
+
+        Sentry.removeExtra("tag")
+        Sentry.removeTag("tag")
```

Comment:
I assume we remove these so it doesn't affect future events. I think this might not be super clear to everyone else, so should we add a comment on top of them to explain this? If we do that, can we do it for `reportException` as well?

Reply:
Also fixed in a34c5b762656e4ed3d17f5e0823dd69a7b1f187b